app/models/firmware_target.rb

```diff
@@ -0,0 +1,33 @@
+class FirmwareTarget < ApplicationRecord
+  has_many :firmware_binary_firmware_targets, :dependent => :destroy
+  has_many :firmware_binaries, :through => :firmware_binary_firmware_targets
+
+  before_create :normalize
+  before_save   :normalize
+
+  # Attributes that need to match for target physical server to be assumed compatible.
+  MATCH_ATTRIBUTES = %i[manufacturer model].freeze
+
+  def normalize
+    manufacturer.downcase!
+    model.downcase!
+  end
+
+  # Firmware targets are nameless, but we need to show something on the UI.
+  def name
+    _('Firmware Target')
+  end
+
+  def to_hash
+    attributes.symbolize_keys.slice(*MATCH_ATTRIBUTES)
```

Comment:
Would it make sense to not store attributes that aren't needed instead of overriding to_hash?

Reply:
We mainly get rid of created/updated timestamps here to get a "pure" hash with relevant attributes only (:manufacturer and :model). Perhaps I should just rename the function to something else?